dockertest/environment.py

```diff
@@ -210,28 +210,42 @@ def __new__(cls, *args, **dargs):
         else:
             _object = ''  # default
 
-        _clues = cls.investigate(str(_object))
-        victims = set(_clues.values())
-        if len(victims) > 1:
-            raise ValueError("Multiple distros detected: %s" % _clues)
-        elif len(victims) != 1:
-            raise ValueError("No distros detected: %s" % _clues)
+        _matches = cls.investigate(str(_object))
+        distro_matches = set(_matches.values())
+        if len(distro_matches) > 1:
+            raise ValueError("Multiple distros detected: %s" % _matches)
+        elif len(distro_matches) != 1:
+            raise ValueError("No distros detected: %s" % _matches)
 
         # immutables can't rely on __init__(), it must be done here
-        new_cd = super(CanonicalDistro, cls).__new__(cls, victims.pop())
-        new_cd.clues = _clues
+        new_cd = super(CanonicalDistro, cls).__new__(cls, distro_matches.pop())
+        new_cd.matches = _matches
         return new_cd
 
+    # Must exactly match the distros list defined above: one method
+    # for each distro. FIXME: define what "hints" is supposed to be.
```

Comment:
Yes, it's kind of like alternate contents in case /etc/os-release isn't parse-able, the filename changed, etc.  TBH, the main reason I added it was to make unittesting easier and preserve the str() interface.

Thinking about it now, it's probably not necessary.  I'm cool with killing it off, if you don't think it useful.  

How about this:  if an initializer-object is passed into ``__new__()``, then we just take that as the distro-name and bypass calling any of the ``distros`` methods.  It'll make unittesting the ``dostros`` methods a bit harder, but nothing we can't figure out.

e.g.
```python


    def __new__(cls, *args, **dargs):
        # Sigh, the str builtin really does use a parameter named 'object' :_(
        if len(args) > 1 or len(dargs) > 1:
            raise TypeError("__init__() takes at most 1 argument (%d given)"
                            % (len(args) + len(dargs)))
        elif len(dargs) == 1:
            if 'object' in dargs:
                _object = dargs['object']
            else:
                raise TypeError("__init__() got unexpected keyword argument %s"
                                % dargs.popitem()[0])
        elif len(args) == 1:
            _object = args[0]
        else:
            _object = ''  # default

        if _object:
            return super(CanonicalDistro, cls).__new__(cls, str(_object))
        else:
            _matches = cls.investigate()
            distro_matches = set(_matches.values())
            if len(distro_matches) > 1:
                raise ValueError("Multiple distros detected: %s" % _matches)
            elif len(distro_matches) != 1:
                raise ValueError("No distros detected: %s" % _matches)
    
            # immutables can't rely on __init__(), it must be done here
            new_cd = super(CanonicalDistro, cls).__new__(cls, distro_matches.pop())
            new_cd.matches = _matches
            return new_cd
```